harvester_e2e_tests/utils.py

```diff
@@ -460,9 +462,10 @@ def _check_vm_ready():
             _check_vm_ready,
             step=5,
             timeout=request.config.getoption('--wait-timeout'))
-    except polling2.TimeoutException as e:
-        errmsg = 'Timed out while waiting for VM to be ready.'
-        raise AssertionError(errmsg) from e
+    except polling2.TimeoutException:
+        errmsg = ('Timed out while waiting for VM to be ready.\n'
+                  f"Stucking in Phase {resp_json['status']['phase']}")
```

Comment:
We don't necessarily need to change this :smile: - this is more of a general comment:

We're leveraging the value of `resp_json.status.phase` from the Python dictionary, we may encounter a `KeyError` if the value doesn't exist which could throw another error while trying to generate `errmsg`.
[we could use the get() functionality on the dict](https://docs.python.org/3/library/stdtypes.html#dict.get)
We may want to try to just set it as something like:
```
f"Sticking in Phase {resp_json.get('status',{}).get('phase', 'default_phase_not_found')}"
```
Where we could create a temporary dict if we can't find the dictionary associated with 'status' and then if we can't find 'phase' to use within the `errmsg` we can just use some default string of some sort.

Reply:
the key will never missed, or it would be a critical issue for the cluster.
you can reference `Line:703` of inner function for more details.